src/shared/globalState.ts

```diff
@@ -1,144 +1,612 @@
-
-const _checkSecretKeysExhaustiveness: CheckSecretKeysExhaustiveness = true
-
-export const GLOBAL_STATE_KEYS = [
-	"apiProvider",
-	"apiModelId",
-	"glamaModelId",
-	"glamaModelInfo",
-	"awsRegion",
-	"awsUseCrossRegionInference",
-	"awsProfile",
-	"awsUseProfile",
-	"awsCustomArn",
-	"vertexKeyFile",
-	"vertexJsonCredentials",
-	"vertexProjectId",
-	"vertexRegion",
-	"lastShownAnnouncementId",
-	"customInstructions",
-	"alwaysAllowReadOnly",
-	"alwaysAllowReadOnlyOutsideWorkspace",
-	"alwaysAllowWrite",
-	"alwaysAllowWriteOutsideWorkspace",
-	"alwaysAllowExecute",
-	"alwaysAllowBrowser",
-	"alwaysAllowMcp",
-	"alwaysAllowModeSwitch",
-	"alwaysAllowSubtasks",
-	"taskHistory",
-	"openAiBaseUrl",
-	"openAiModelId",
-	"openAiCustomModelInfo",
-	"openAiUseAzure",
-	"ollamaModelId",
-	"ollamaBaseUrl",
-	"lmStudioModelId",
-	"lmStudioBaseUrl",
-	"anthropicBaseUrl",
-	"modelMaxThinkingTokens",
-	"azureApiVersion",
-	"openAiStreamingEnabled",
-	"openAiR1FormatEnabled",
-	"openRouterModelId",
-	"openRouterModelInfo",
-	"openRouterBaseUrl",
-	"openRouterSpecificProvider",
-	"openRouterUseMiddleOutTransform",
-	"googleGeminiBaseUrl",
-	"allowedCommands",
-	"soundEnabled",
-	"ttsEnabled",
-	"ttsSpeed",
-	"soundVolume",
-	"diffEnabled",
-	"enableCheckpoints",
-	"checkpointStorage",
-	"browserViewportSize",
-	"screenshotQuality",
-	"remoteBrowserHost",
-	"fuzzyMatchThreshold",
-	"writeDelayMs",
-	"terminalOutputLineLimit",
-	"terminalShellIntegrationTimeout",
-	"mcpEnabled",
-	"enableMcpServerCreation",
-	"alwaysApproveResubmit",
-	"requestDelaySeconds",
-	"rateLimitSeconds",
-	"currentApiConfigName",
-	"listApiConfigMeta",
-	"vsCodeLmModelSelector",
-	"mode",
-	"modeApiConfigs",
-	"customModePrompts",
-	"customSupportPrompts",
-	"enhancementApiConfigId",
-	"experiments", // Map of experiment IDs to their enabled state.
-	"autoApprovalEnabled",
-	"customModes", // Array of custom modes.
-	"unboundModelId",
-	"requestyModelId",
-	"requestyModelInfo",
-	"unboundModelInfo",
-	"modelTemperature",
-	"modelMaxTokens",
-	"mistralCodestralUrl",
-	"maxOpenTabsContext",
-	"browserToolEnabled",
-	"lmStudioSpeculativeDecodingEnabled",
-	"lmStudioDraftModelId",
-	"telemetrySetting",
-	"showRooIgnoredFiles",
-	"remoteBrowserEnabled",
-	"language",
-	"maxWorkspaceFiles",
-	"maxReadFileLine",
-	"fakeAi",
-] as const
+const providerNames: Record<ProviderName, true> = {
+	anthropic: true,
+	glama: true,
+	openrouter: true,
+	bedrock: true,
+	vertex: true,
+	openai: true,
+	ollama: true,
+	lmstudio: true,
+	gemini: true,
+	"openai-native": true,
+	deepseek: true,
+	"vscode-lm": true,
+	mistral: true,
+	unbound: true,
+	requesty: true,
+	"human-relay": true,
+	"fake-ai": true,
+}
 
-export const PASS_THROUGH_STATE_KEYS = ["taskHistory"] as const
+const PROVIDER_NAMES = Object.keys(providerNames) as ProviderName[]
+
+const providerNamesEnum: [ProviderName, ...ProviderName[]] = [
+	PROVIDER_NAMES[0],
+	...PROVIDER_NAMES.slice(1).map((p) => p),
+]
```

Comment:
You'll see this technique employed in a few places. This a trick that allows you to convert a typescript string union type to an array that can be passed to `zod.enum`. I've named all of these to reflect this.